src/_pytest/config/__init__.py

```diff
@@ -744,9 +746,9 @@ def consider_pluginarg(self, arg: str) -> None:
                     del self._name2plugin["pytest_" + name]
             self.import_plugin(arg, consider_entry_points=True)
 
-    def consider_conftest(self, conftestmodule: types.ModuleType) -> None:
+    def consider_conftest(self, conftestmodule: types.ModuleType, name: str) -> None:
```

Comment:
strictly speaking this one is a "breaking api change" do we have any idea on the impact - i suspect its zero, but i'm not certain

Reply:
Would it fix to have `name` default to `None` and use `conftestmodule.__file__` (maybe normalized ) if so ?

Or, another way I considered would be to run `mod = import_path(conftestpath, mode=importmode, root=rootpath)` first, then using `mod.__file__` to pass the `existing` check rather than `str(conftestpath)`: that works too, with the advantage of not changing the api, and not altering the keys in `name2plugins` compared to current. The downside is that `import_path` is now also called for plugins that are already registered (which might not be important since the module should be already imported so the `import_module` will do nothing ? but I wasn't sure of that).

Reply:
The first way sounds practical, 

I'd recommend normalizing

I'd defer implementation, in Case  there's next to no impact it should go into 8.x as is and the backport would trigger a deprecation warning plus use none